### ISSUE TITLE
List block toolbar active state for unordered vs. ordered is wonky

### ISSUE THREAD

**Issue by thatdevgirl** [WordPress/gutenberg#15250]
**Describe the bug**
When changing the list block between an unordered list and ordered list, the active state for the selected list type does not change.

**To reproduce**
1. Create a new post and add the OOTB Gutenberg list block.
2. Add several items to the list block.
3. In the toolbar, select the ordered list option. Note that the list changes from unordered to ordered, but the unordered list option in the toolbar is still selected.
4. Click out of the list block in the editor to deselect it.
5. Click back into the list block to select it. Note that the ordered list option in the toolbar is now selected.
6. Select the unordered list option. Note that the list changes to unordered, but the ordered list option in the toolbar is still selected.
7. Again, click out and then back into the list block. Note that the unordered list option in the toolbar is now selected.

**Expected behavior**
I expect the selected list option to have an active state when it is clicked on, without having to deselect and re-select that block.

**Desktop (please complete the following information):**
 - OS: MacOS 10.14.4
 - Browser: Chrome (latest)

**Additional context**
- WP core v. 5.1.1

**Comment by Jackie6** [comment]
What is your current version of Gutenberg? I encountered this bug in v. 5.4 but v. 5.5 has fixed it.

**Comment by thatdevgirl** [comment]
It is whatever version is baked into WP core 5.1.1

**Comment by swissspidy** [comment]
@thatdevgirl Still happening with WordPress 5.2.1?

**Comment by thatdevgirl** [comment]
@swissspidy Somewhat. The bug is different now. After I add the list block and add several items to the list, the bug occurs only the first time I switch list types. See attached gif, but I outlined the steps to reproduce below:

1. Click into the list block.
2. Switch from unordered to ordered list in the tool bar. The list changes but the unordered icon in the toolbar is still selected.
3. Switch back from ordered to unordered list. The list changes and the unordered icon is still (correctly now) selected.)
4. Switch back from unordered list to ordered list. Both the list and the toolbar icon change.

I tested this on WP 5.2.1 with no plugins installed on the latest version of Chrome.

![list-type-bug](https://user-images.githubusercontent.com/5687778/58375084-0f5a5900-7f19-11e9-9767-532d44623eba.gif)
